doc/chapters/developers/index.rst

```diff
@@ -0,0 +1,145 @@
+
+.. _developers:
+
+Developer guidelines
+********************
+
+This section describes the internal structure of SoftwareContainer, and the major design considerations.
+
+The intended reader is a developer that wants to understand and work with the internals of SoftwareContainer,
+or anyone that simply wants a better insight for e.g. troubleshooting.
+
+.. todo::
+    The diagrams in this chapter does not differ between sync and async calls, should be fixed
+
+
+Important internal interfaces
+=============================
+
+Interface: libsoftwarecontainer.h
+---------------------------------
+
+This interface is what the Agent uses to manage the life cycle of container instances, e.g. creation and
+destruction. The interface provides access to the underlying ``ContainerAbstractInterface`` so users can
+access the container interface. The interface is implemented by ``SoftwareContainerLib``.
```

Comment:
Maybe something like "which users can use to access the container"  instead of "so users can access...."